### PR TITLE
[SMTChecker] Fix ICE in inherited state var

### LINKED ISSUE #6562: [SMTChecker] Support state variables of base contracts
BECToken from https://github.com/ConsenSys/mythril-classic/tree/develop/solidity_examples gives
```
ibsolidity/formal/SMTChecker.cpp(1622): Throw in function dev::solidity::smt::Expression dev::solidity::SMTChecker::currentValue(const dev::solidity::VariableDeclaration&)
Dynamic exception type: boost::wrapexcept<langutil::InternalCompilerError>
std::exception::what: 
[dev::tag_comment*] = 
```

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes #6562

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -4,9 +4,11 @@ Language Features:
 
 
 Compiler Features:
+ * SMTChecker: Support inherited state variables.
 
 
 Bugfixes:
+ * SMTChecker: Fix ICE when visiting state variable inherited from base class.
```
Comment (chriseth) [inline Changelog.md]:
```suggestion
 * SMTChecker: Fix internal error when visiting state variable inherited from base class.
```

File: libsolidity/formal/SMTChecker.cpp [hunk 2]
```diff
@@ -78,8 +78,10 @@ void SMTChecker::analyze(SourceUnit const& _source, shared_ptr<Scanner> const& _
 
 bool SMTChecker::visit(ContractDefinition const& _contract)
 {
-	for (auto _var : _contract.stateVariables())
-		createVariable(*_var);
+	for (auto const& contract: _contract.annotation().linearizedBaseContracts)
+		for (auto var : contract->stateVariables())
+			if (var->isVisibleInDerivedContracts())
```
Comment (chriseth) [inline libsolidity/formal/SMTChecker.cpp]: Doesn't this exclude the private variables from the current contract?

Comment (chriseth) [reply libsolidity/formal/SMTChecker.cpp]: You might want to use ContractType::stateVariables

Comment (chriseth) [reply libsolidity/formal/SMTChecker.cpp]: Yeah ok, perhaps it's not that useful.

Comment (leonardoalt) [reply libsolidity/formal/SMTChecker.cpp]: Oh true, was thinking only about base classes. Will update

Comment (leonardoalt) [reply libsolidity/formal/SMTChecker.cpp]: and add a test

**Review by chriseth — APPROVED** [review]

**Pull request merged**